Lib/_pydatetime.py

```diff
@@ -1755,7 +1773,7 @@ class datetime(date):
     __slots__ = time.__slots__
 
     def __new__(cls, year, month=None, day=None, hour=0, minute=0, second=0,
-                microsecond=0, tzinfo=None, *, fold=0):
+                microsecond=0, tzinfo=None, *, fold=0, submicrosecond=0):
```

Comment:
Why did you choose to name this new parameter "submicrosecond" specifically? The definition of what "sub" means in this context seems arbitrary and I have not seen this term used anywhere else. The natural term would be "nanosecond" like in the PR title.

There has also been some unresolved talk about the Python API in the related PR https://github.com/python/cpython/pull/21987.

Reply:
https://github.com/python/cpython/pull/92078#discussion_r866860074